gripper/simulator/main.cpp

```diff
@@ -18,33 +25,65 @@ void signal_handler(int signum) {
 /**
  * The CAN bus.
  */
-static auto canbus = can::sim::bus::SimCANBus(can::sim::transport::create());
 
 static auto capsensor = fdc1004_simulator::FDC1004{};
-static auto sim_eeprom = eeprom::simulator::EEProm{};
 static auto sensor_map =
     i2c::hardware::SimI2C::DeviceMap{{capsensor.get_address(), capsensor}};
 static auto i2c2 = i2c::hardware::SimI2C{sensor_map};
-static auto i2c_device_map =
-    i2c::hardware::SimI2C::DeviceMap{{sim_eeprom.get_address(), sim_eeprom}};
-static auto i2c3 = i2c::hardware::SimI2C{i2c_device_map};
 
 static test_mocks::MockSensorHardware fake_sensor_hw{};
 
-int main() {
+auto handle_options(int argc, char** argv) -> po::variables_map {
+    auto cmdlinedesc =
+        po::options_description("simulator for the OT-3 gripper");
+    auto envdesc = po::options_description("");
+    cmdlinedesc.add_options()("help,h", "Show this help message.");
+    auto can_arg_xform = can::sim::transport::add_options(cmdlinedesc, envdesc);
+    auto eeprom_arg_xform =
+        eeprom::simulator::EEProm::add_options(cmdlinedesc, envdesc);
+
+    po::variables_map vm;
+    po::store(po::parse_command_line(argc, argv, cmdlinedesc), vm);
+    if (vm.count("help")) {
+        std::cout << cmdlinedesc << std::endl;
+        std::exit(0);
+    }
+    po::store(po::parse_environment(
+                  envdesc,
+                  [can_arg_xform, eeprom_arg_xform](
+                      const std::string& input_val) -> std::string {
+                      auto can_xformed = can_arg_xform(input_val);
+                      if (can_xformed != "") {
+                          return can_xformed;
+                      }
+                      auto eeprom_xformed = eeprom_arg_xform(input_val);
+                      return eeprom_xformed;
+                  }),
+              vm);
+    po::notify(vm);
+    return vm;
+}
+
+int main(int argc, char** argv) {
     signal(SIGINT, signal_handler);
 
     LOG_INIT("GRIPPER", []() -> const char* {
         return pcTaskGetName(xTaskGetCurrentTaskHandle());
     });
-
+    auto options = handle_options(argc, argv);
+    auto sim_eeprom = std::make_shared<eeprom::simulator::EEProm>(options);
+    auto i2c_device_map = i2c::hardware::SimI2C::DeviceMap{
+        {sim_eeprom->get_address(), *sim_eeprom}};
+    auto i2c3 = std::make_shared<i2c::hardware::SimI2C>(i2c_device_map);
```

Comment:
what is the benefit of making these shared pointers here?

Reply:
Before this PR, all these objects could be statically created as globals. Now, some of them depend on command line options, so they need to be created in `main`. We could create them on the stack but that's usually bad practice when you're writing a general purpose program, so let's put them on the heap. Then, if we're putting it on the heap, we should really have it in a smart pointer. So, shared ptrs